obspy/io/nordic/core.py

```diff
@@ -660,6 +661,118 @@ def _read_picks(tagged_lines, new_event):
+    #   3. Assure that it's really Old Nordic format
+    #     3.a  Check the full format of the pick-time
+    #     3.b  Check phase-descriptor
+
+    # Let's  assume it's a New Nordic format for now.
+    is_phase = False
+    nordic_format = 'UKN'
+
+    for line in pickline:
+        # if whole line is blank, it cannot be a phase line
+        if line.isspace():
+            return nordic_format, is_phase
+
+        old_format_secs = line[23:28].strip(' ')
+        # Check whether the old-formatted seconds are a float rather than int.
+        # If they are int, then it is probably new nordic format.
+        try:
+            if str(int(old_format_secs.replace(' ', '').replace('A', '')))\
+                    == old_format_secs:
+                nordic_format = 'NEW'
+        except ValueError:
+            nordic_format = 'OLD'
+            pass
+
+        if nordic_format == 'NEW' or nordic_format == 'UKN':
+            hr = int(line[26:28].strip() or 0)
+            min = int(line[28:30].strip() or 0)
+            sec = float(line[30:37].strip() or 0)
+            if (line[9:10] == ' ' and
+               line[14:15] == ' ' and
+               line[26:27] in ' 012' and
+               hr >= 0 and hr <= 26 and
+               min >= 0 and min <= 60 and
+               sec >= 0.0 and sec <= 200 and
+               not line[28:37].isspace()):
+                if (_nordic_iasp_phase_ok(line[16:25]) or
+                        line[16:19] == 'END' or
+                        line[16:19] == 'BAZ' or
+                        line[16:18] == 'IA' or
+                        line[16:18] == 'IV' or
+                        line[16:18] == 'AM' or
+                        line[16:18] == 'AP' or
+                        (line[16:17] == ' ' and
+                         line[15:16] == 'I' or line[15:16] == 'E')):
+                    is_phase = True
+                    nordic_format = 'NEW'
+                    return nordic_format, is_phase
+        else:
+            hr = int(line[18:20].strip() or 0)
+            min = int(line[20:22].strip() or 0)
+            sec = float(line[22:28].strip() or 0)
+            if (line[18:19] in ' 012' and
+               hr >= 0 and hr <= 26 and
+               min >= 0 and min <= 60 and
+               sec >= 0.0 and sec <= 200):
+                if (_nordic_iasp_phase_ok(line[10:18]) or
+                        line[10:13] == 'END' or
+                        line[10:13] == 'BAZ' or
+                        line[10:12] == 'IA' or
+                        line[10:12] == 'IV' or
+                        line[10:12] == 'AM' or
+                        line[10:12] == 'AP' or
+                        (line[10:11] == ' ' and
+                         line[9:10] == 'I' or line[9:10] == 'E')):
+                    is_phase = True
+                    nordic_format = 'OLD'
+                    return nordic_format, is_phase
+    # If neither New nor Old format, return "UKN"
+    return nordic_format, is_phase
```

Comment:
There could be a pick with no phase information (not even "E" or "I"). Then the function correctly returns `is_phase=False`, but it can still sense the new vs old Nordic version based on the time in the line. Not that that's a common occurrence, but with files from old systems I don't want to rule out ever encountering a strange edge case.

But you're right that the `is_phase`-command is not currently made use of anywhere else...

Reply:
Ah yup, sorry, I missed the bit in lines 714-718 - I thought `nordic_format` was only ever getting reset at line 767 and 746. :+1: